src/main/scala/sbtdatabricks/DatabricksPlugin.scala

```diff
@@ -64,29 +64,33 @@ object DatabricksPlugin extends AutoPlugin {
       }
     }.flatMap(c => c)
   }
-  
+
   private lazy val dbcClasspath = Def.task {
     ((Keys.`package` in Compile).value +: (managedClasspath in Runtime).value.files
       ).filterNot(_.getName startsWith "scala-")
   }
 
   private val dbcFetchClusters = taskKey[Seq[Cluster]]("Fetch all available clusters.")
-  
+
   private lazy val uploadImpl: Def.Initialize[Task[Seq[UploadedLibrary]]] = Def.taskDyn {
     val client = dbcApiClient.value
     val folder = dbcLibraryPath.value
+    val existing = existingLibraries.value
+    val existingSnapshots = existing.filter(_.name.contains("-SNAPSHOT"))
     val classpath = dbcClasspath.value
     var deleteMethodFinished = false
-    val deleteMethod = client.deleteLibraries(existingLibraries.value)
+    val deleteMethod = client.deleteLibraries(existingSnapshots)
     deleteMethodFinished ||= deleteMethod
     if (deleteMethodFinished) {
       Def.task {
-        val uploaded = classpath.map { jar =>
+        // Either upload the newer SNAPSHOT versions, or everything, because they don't exist yet.
+        val toUpload = classpath.toSet -- existing.map(_.jar) ++ existingSnapshots.map(_.jar)
```

Comment:
This problem exists in the current state as well. I'll try to come up with a solution in another PR. The problem with artifact naming is that it has no information about the groupId. It's in the format `$artifactId-$version.jar`. If I try to delete by artifactId only, there may be conflicts (for example there are 108 projects with the artifactId `core`). The easiest way to handle that is through adding all dependencies as a bundle, or with their maven coordinates, and handle resolution in DBC, which handles such incompatibilities.

For now, is manually deleting the older version through the UI a huge problem?